aerogear/mobile-security-service
# feat(db): Adding device table to the database (AEROGEAR-8533)

## Motivation
Unable to know when a new client inits for the first time without storing at least device id in the database. 
https://issues.jboss.org/browse/AEROGEAR-8533

## What
Adding a device table to the database with a 1-many relationship from versions.

## Why
Needed to differentiate devices first time connecting to capture the number of clients using a specific version of an app.

## How
Adding the device table

## Verification Steps
1. Remove any existing docker Postgres containers (`docker rm <container_id>`)
2. Run `docker-compose up db`
3. Run `go run cmd/mobile-security-service/main.go`
3. Run `docker exec -it <postgres-container-id> bash`
4. Run `psql -U postgres`
5. Run `\c mobile_security_service` - This connects to our database
6 Run `\dt` - Lists all tables in the current database
7 Verify the device tables exists

Note: You can also use pgAdmin or some other db management tool to verify the tables are setup correctly

## Checklist:


## Progress
